microsoft/testsuites/stress/stress_ng_suite.py

```diff
@@ -0,0 +1,165 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+from pathlib import Path, PurePath
+from typing import Any, Dict, List, cast
+
+from lisa import (
+    Environment,
+    Logger,
+    Node,
+    RemoteNode,
+    TestCaseMetadata,
+    TestSuite,
+    TestSuiteMetadata,
+    notifier,
+)
+from lisa.messages import SubTestMessage, TestStatus, create_test_result_message
+from lisa.testsuite import TestResult
+from lisa.tools import StressNg
+from lisa.util import SkippedException
+from lisa.util.process import Process
+
+
+@TestSuiteMetadata(
+    area="stress",
+    category="stress-ng",
+    description="""
+    A suite for running the various classes of stressors provided
+    by stress-ng.
+    """,
+)
+class StressNgTestSuite(TestSuite):
+    TIME_OUT = 3600
+    CONFIG_VARIABLE = "stress_ng_jobs"
+
+    @TestCaseMetadata(
+        description="""
+        """,
+        timeout=TIME_OUT,
+        priority=4,
+    )
+    def verify_stress_ng_jobfile(
+        self,
+        log: Logger,
+        node: Node,
+        variables: Dict[str, Any],
+        environment: Environment,
+        log_path: Path,
+        result: TestResult,
+    ) -> None:
+        if self.CONFIG_VARIABLE in variables:
+            jobs = variables[self.CONFIG_VARIABLE]
+            for job_file in jobs:
+                self._run_stress_ng_job(job_file, environment, result)
+        else:
+            raise SkippedException("No jobfile provided for stress-ng")
+
+    @TestCaseMetadata(
+        description="Runs stress-ng's 'cpu' class stressors",
+        priority=4,
+        timeout=3600,
```

Comment:
Timeout used by LISA to force stop a case. But it makes the case doesn't have a right result. Make sure the test case end before the timeout. The time to run should be provided in description too.

Reply:
These tests are finished with 3600s.